pkg/controller/common/cni/cni_test.go

```diff
@@ -0,0 +1,32 @@
+package cni
+
+import (
+	"testing"
+
+	"github.com/maistra/istio-operator/pkg/controller/common"
+	"github.com/maistra/istio-operator/pkg/controller/common/test/assert"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+func TestInitConfig_disablingCNI(t *testing.T) {
+	operatorNamespace := "istio-operator"
+	InitializeGlobals(operatorNamespace)()
+	var m manager.Manager
+	config, err := InitConfig(m)
+	assert.Equals(err, nil, "", t)
+	assert.Equals(config.Enabled, false, "", t)
+
+	// Quick test cleanup
+	common.Config.OLM.CNIEnabled = true
```

Comment:
It would be safer to store the value of `CNIEnabled` in a local variable and reset it to that stored value instead of setting it to `true`. This way this test never has any side-effects.

Reply:
I don't think I completely understand.

The point of this test function would be to read the config value into the `common` struct, decoupling it into a local variable does not seem like a logical step to me.

Care to elaborate on this?

Reply:
In this line here, you clean things up by setting `CNIEnabled` to `true`. But how do you know it had that value when you started the test? Right now, it's true that the value is `true`, but this could change in the future. If we ever change this to default to `false`, running this test will change that value and possibly affect any tests that run after this test. 

I know this is all hypothetical and it's just me nit-picking, but I've seen these sorts of things happen too often and they always require too much time to find the problem.

Reply:
Gotcha, thanks for explaining that side. I totally agree. Fixing.